.github/workflows/ci-generated.yml

```diff
@@ -18,6 +18,29 @@ jobs:
       uses: actions/setup-python@v1
       with:
         python-version: "3.7"
+    - name: Setup Git Profile
+      run: |
+        git config --local user.email "action@github.com"
+        git config --local user.name "GitHub Action"
+    - name: Revert Previous Change and Merge Master
+      run: |
+        git ls-remote origin
+        if git ls-remote origin | grep refs/ci_generated/master; then
+          # If the remote hidden branch exists
+          # Map the remote ref to branch.
+          git fetch origin +refs/ci_generated/master:refs/remotes/origin/ci_generated/master
+          git checkout -b ci_generated/master origin/ci_generated/master
+
+          # Revert the previous generated files, so that the merge is clean.
+          git revert HEAD
+
+          # Merge master.
+          MASTER_REV=$(git log -1 master --pretty=%H)
+          git merge master -m "Merge master ${MASTER_REV}"
```

Comment:
This code will take the code from the `ci_generated` branch, and then merge `master` into this branch.
While in most cases this should not fail, there is one way to ensure that by doing the opposite.

```
git checkout -b ci_generated/master origin/master
get merge ci_generated/master -m "Merge master ${MASTER_REV}" -s ours --allow-unrelated-histories
```

The difference would be the order in which the parents of the branch are listed, but this way we can always use the `ours` strategy to completely ignore the previously generated files from old `ci_generated` branches, and this should always work, even if we were to force-push the master branch one day.